tls/s2n_config.c

```diff
@@ -308,28 +308,10 @@ struct s2n_config *s2n_config_new(void)
     return new_config;
 }
 
-static int s2n_config_store_ticket_key_comparator(const void *a, const void *b)
-{
-    if (((const struct s2n_ticket_key *) a)->intro_timestamp >= ((const struct s2n_ticket_key *) b)->intro_timestamp) {
-        return S2N_GREATER_OR_EQUAL;
-    } else {
-        return S2N_LESS_THAN;
-    }
-}
```

Comment:
Potential subtle problem related to https://github.com/aws/s2n-tls/pull/5146#discussion_r1972105455: the set enforced the intro_timestamp ordering. The array uses calls to s2n_config_add_ticket_crypto_key, which can set an intro time in the future, in the past, whatever. So when removing an expired key from the array, the expired key could actually be somewhere in the middle. Since s2n_set_len / s2n_array_num_elements go off of the basic memory size of the list, not considering deleted / null elements, the apparent "length" of the key array won't change after a removal of an element from the middle. That seems like an issue.